dojo/unittests/tools/test_trustwave_fusion_api_parser.py

```diff
@@ -62,6 +62,7 @@ def test_parse_file_with_multiple_vuln_has_multiple_findings(self):
         self.assertEqual("Info", finding.severity)
         self.assertIsNone(finding.cve)  # should be none since CVE is "CVE-NO-MATCH"
         self.assertIsNone(finding.component_name)
+        self.assertIsNone(finding.component_version)
```

Comment:
do the assertEqual here.
`self.assertEqual("comp", finding.component_name)`

Reply:
There is no comp name here though: the parser will detect the applicationName (in the location field of the JSON file) being "None," so it will pick nothing up. Thus, the field here should be None. Am I missing something here though?

Reply:
Sorry maybe not the best finding.
I saw another finding in the report that have component data.

Reply:
if you take a look at:
https://github.com/DefectDojo/django-DefectDojo/blob/77bd0431df417ad1ca57cbd8447cee841a01e0a4/dojo/unittests/scans/trustwave_fusion_api/test_cve.json#L17
I can see that you have component name+version as CPE string.
You can do something like this:
```
             if "applicationCpe" in vuln["location"]:
                    cpe = CPE(vuln["location"]["applicationCpe"])
                    component_name = cpe.get_vendor()[0] + ':' if len(cpe.get_vendor()) > 0 else ''
                    component_name += cpe.get_product()[0] if len(cpe.get_product()) > 0 else ''
                    finding.component_name = component_name if component_name else None
                    finding.component_version = cpe.get_version()[0] if len(cpe.get_version()) > 0 else None
```

Reply:
I see it now, thanks!

Reply:
Done!